### PR TITLE
Testnet params & genesis-nodejs cleanup

### PR DESCRIPTION
**Pull request by akshaynexus**

### CODE REVIEW

File: src/chainparams.cpp [hunk 1]
```diff
@@ -647,48 +606,44 @@ class CTestNetParams : public CChainParams {
...
+        pchMessageStart[3] = 0x7u;
+        nDefaultPort = 51928;
         nBIP44ID = (int)WithHardenedBit(1);
 
         nModifierInterval = 10 * 60;    // 10 minutes
         nStakeMinConfirmations = 225;   // 225 * 2 minutes
-        nTargetSpacing = 120;           // 2 minutes
+        nTargetSpacing = 60;           // 2 minutes
         nTargetTimespan = 24 * 60;      // 24 mins
+        nBlockReward = 3 * COIN;
```
Comment (TheQuantumPhysicist) [inline src/chainparams.cpp]: Please make this 2

File: src/chainparams.cpp [hunk 2]
```diff
@@ -509,9 +468,9 @@ class CMainParams : public CChainParams {
 
 
         vDevFundSettings.emplace_back(0,
-            DevFundSettings("RJAPhgckEgRGVPZa9WoGSWW24spskSfLTQ", 10, 60));
+            DevFundSettings("RJAPhgckEgRGVPZa9WoGSWW24spskSfLTQ", 10, 60,0));
```
Comment (TheQuantumPhysicist) [inline src/chainparams.cpp]: Always space after comman

File: src/chainparams.cpp (hunk 1 above)
Comment (akshaynexus) [reply src/chainparams.cpp]: So then the split would be 1 coin to devaddr and one to staker.in that case

File: src/chainparams.cpp (hunk 2 above)
Comment (akshaynexus) [reply src/chainparams.cpp]: Ok will do

File: src/chainparams.cpp (hunk 1 above)
Comment (akshaynexus) [reply src/chainparams.cpp]: Dismissed after discussion

File: src/pow.cpp [hunk 3]
```diff
@@ -113,11 +113,11 @@ bool CheckProofOfWork(uint256 hash, unsigned int nBits, const Consensus::Params&
 
     // Check range
     if (fNegative || bnTarget == 0 || fOverflow || bnTarget > bnProofOfWorkLimit)
-        return false;
+        return error("Out of range");
```
Comment (TheQuantumPhysicist) [inline src/pow.cpp]: No need for this now. So either remove it or put a TODO: GHOSTFORK on it.

File: src/pow.cpp [hunk 4]
```diff
@@ -113,11 +113,11 @@ bool CheckProofOfWork(uint256 hash, unsigned int nBits, const Consensus::Params&
...
 
     // Check proof of work matches claimed amount
     if (UintToArith256(hash) > bnTarget)
-        return false;
+        return error("hash doesnt match target");
```
Comment (TheQuantumPhysicist) [inline src/pow.cpp]: This too

File: src/pow.cpp [hunk 5]
```diff
@@ -98,7 +98,7 @@ bool CheckProofOfWork(uint256 hash, unsigned int nBits, const Consensus::Params&
     {
         arith_uint256 nMinProofOfWorkLimit = arith_uint256("00ffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff");
         arith_uint256 nMaxProofOfWorkLimit = UintToArith256(params.powLimit);
-        arith_uint256 nStep = ((nMaxProofOfWorkLimit - nMinProofOfWorkLimit) / nLastImportHeight);
+        arith_uint256 nStep = nMaxProofOfWorkLimit - nMinProofOfWorkLimit;
         bnProofOfWorkLimit = nMinProofOfWorkLimit + nStep * nBlockHeight;
```
Comment (TheQuantumPhysicist) [inline src/pow.cpp]: Remove nStep*nBlockHeight

File: src/chainparams.cpp [hunk 6]
```diff
@@ -482,7 +450,8 @@ class CMainParams : public CChainParams {
         nStakeMinConfirmations = 225;   // 225 * 2 minutes
         nTargetSpacing = 120;           // 2 minutes
         nTargetTimespan = 24 * 60;      // 24 mins
-
+        nBlockReward = 2 * COIN;
```
Comment (TheQuantumPhysicist) [inline src/chainparams.cpp]: Should be 6

**Pull request merged**